### PR TITLE
Create Jenkinsfile

### PR DESCRIPTION
**Pull request by miklis**
## Status
**READY**

## Description
Replace static Jenkins job with Jenkinsfile.
This should cover now ~ all existing functionality from previous job.

### CODE REVIEW

**Comment by jupe** [comment]
please @miklis update this a bit,,,

**Comment by marhil01** [comment]
Why this not merged yet?

**Comment by jupe** [comment]
@marhil01  it's not ready - doesn't cover even half of current Jenkins job + haven't had time to finish it yet..

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,220 @@
+echo "Start to build"
+
+properties ([
+    buildDiscarder(
+        logRotator(
+            artifactNumToKeepStr: '10',
+            numToKeepStr: '100'
+        )
+    )
```
[the remark this answers is not in the document]
Comment (jupe) [reply Jenkinsfile]: is it really needed here since this doesn't use any limited HW resources.. ?

[the remark this answers is not in the document]
Comment (YifeiZuo01) [reply Jenkinsfile]: true, not necessary, it takes more slaves, but shorten time for debugging.

**Pull request merged**